conda_forge_tick/chained_db.py

```diff
@@ -0,0 +1,72 @@
+"""Base class for chaining DBs
+
+ChainDBSingleton
+Copyright 2015-2016, the xonsh developers
+"""
+
+import itertools
+
+from collections import ChainMap
+from collections.abc import MutableMapping
+
+
+class ChainDBSingleton(object):
+    """Singleton for representing when no default value is given."""
+
+    __inst = None
+
+    def __new__(cls):
+        if ChainDBSingleton.__inst is None:
+            ChainDBSingleton.__inst = object.__new__(cls)
+        return ChainDBSingleton.__inst
+
+
+Singleton = ChainDBSingleton()
+
+
+class ChainDB(ChainMap):
+    """ A ChainMap who's ``_getitem__`` returns either a ChainDB or
+    the result"""
+
+    def __getitem__(self, key):
+        res = None
+        results = []
+        # Try to get all the data from all the mappings
+        for mapping in self.maps:
+            results.append(mapping.get(key, Singleton))
+        # if all the results are mapping create a ChainDB
+        if all([isinstance(result, MutableMapping) for result in results]):
+            for result in results:
+                if res is None:
+                    res = ChainDB(result)
+                else:
+                    res.maps.append(result)
+        elif all([isinstance(result, list) for result in results]):
```

Comment:
Should this really be list, and not non-string sequence?

Reply:
I think the main issue with this is that tuples might not make sense when extended.

Reply:
Ahh I see ok. So really, you could test against MutableSequence and be fine (since strings are immutable)